python/torch_mlir/csrc/base_lazy_backend/shape_inference.cpp

```diff
@@ -42,6 +42,12 @@ std::vector<torch::lazy::Shape> compute_shape_hardtanh(
   return {Shape(self.scalar_type(), self.sizes().vec())};
 }
 
+std::vector<torch::lazy::Shape> compute_shape_where(
```

Comment:
Is there a reason we are not upstreaming these? This is supposed to be temporary per the TODO above -- is there a plan to upstream these?

Reply:
the TS backend does not support the `where` op and thus we can't upstream it. Only when the TS backend does support the `where` op would we consider upstreaming it. Or better yet, if they do provide support for the `where` op, then this will probably mean that they already have an implementation for `compute_shape_where` and we can deprecate this one

Reply:
Originally the plan was to upstream these functions. But in PyTorch they are now working on transition to structured kernels which should replace all this custom shape functions. So I believe that in nearest future we will just remove all these functions.